.teamcity/_self/lib/customBuildType/E2EBuildType.kt

```diff
@@ -0,0 +1,170 @@
+import _self.bashNodeScript
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildSteps
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildType
+import jetbrains.buildServer.configs.kotlin.v2019_2.ParametrizedWithType
+import jetbrains.buildServer.configs.kotlin.v2019_2.Project
+import jetbrains.buildServer.configs.kotlin.v2019_2.BuildFeatures
+import jetbrains.buildServer.configs.kotlin.v2019_2.Triggers
+import jetbrains.buildServer.configs.kotlin.v2019_2.Dependencies
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.PullRequests
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.commitStatusPublisher
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.notifications
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.perfmon
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildFeatures.pullRequests
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.ScriptBuildStep
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.dockerCommand
+import jetbrains.buildServer.configs.kotlin.v2019_2.buildSteps.script
+import jetbrains.buildServer.configs.kotlin.v2019_2.failureConditions.BuildFailureOnMetric
+import jetbrains.buildServer.configs.kotlin.v2019_2.failureConditions.failOnMetricChange
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.schedule
+import jetbrains.buildServer.configs.kotlin.v2019_2.triggers.vcs
+
+/**
+ * This class provides a template to easily create end-to-end build configurations.
+ *
+ * To use, import then invoke the `E2EBuildType` class instance with the parameters
+ * required, as outlined in the class definition.
+ *
+ * @param buildId Human-readable value, single string.
+ * @param buildUuid UUID value generated by TeamCity and obtainable via the web UI.
+ * 		Refer to https://fieldguide.automattic.com/quality-engineering-in-calypso/running-tests/.
+ * @param buildName Human-readable display name. Primarily used in the GUI.
+ * @param buidDescription Human-readable description. Primarily used in the GUI.
+ * @param concurrentBuilds Set the number of simultaneous builds. Defaults to 0 (infinite).
+ * @param getCalypsoLiveURL String containing the commands to be run in order to obtain the live URL. Only used for Calypso E2E.
+ * @param testGroup String corresponding to an existing group defined for Jest Runner Group.
+ * @param buildParams Environment variables and other parameters to set for the build.
+ * @param buildFeatures Features to enable on top of the default feature set (perfmon, commitStatusPublisher).
+ * @param buildTriggers Rules to trigger the build. By default, no triggers are defined.
+ * @param buildDepedencies If the build configuration depends on another existing build configuration, define it here.
+ */
+open class E2EBuildType(
+	var buildId: String,
+	var buildUuid: String,
+	var buildName: String,
+	var buildDescription: String,
+	var concurrentBuilds: Int = 0,
```

Comment:
The only user of this param is the Pre-Release Tests, which has a maximum concurrent build count of 1.